xDrip/Scenes/Settings/SettingsMode/SettingsModeRoot/SettingsModeRootViewController.swift

```diff
@@ -64,8 +70,81 @@ class SettingsModeRootViewController: NibViewController, SettingsModeRootDisplay
         interactor?.doLoad(request: request)
     }
     
+    private func setupUI() {
+        title = "settings_mode_settings_title".localized
+        
+        let masterViewController = SettingsModeMasterViewController()
+        let followerViewController = SettingsModeFollowerViewController()
+        
+        addChild(masterViewController)
+        addChild(followerViewController)
+        
+        self.masterViewController = masterViewController
+        self.followerViewController = followerViewController
+        
+        let titles = UserDeviceMode.allCases.map({ $0.title })
+        segmentedControl.removeAllSegments()
+        titles.forEach {
+            segmentedControl.insertSegment(
+                withTitle: $0,
+                at: segmentedControl.numberOfSegments,
+                animated: false
+            )
+        }
+        
+        segmentedControl.selectedSegmentIndex = User.current.settings.deviceMode.rawValue
+        onSegmentedControlValueChanged(segmentedControl)
+    }
+    
+    private func setMasterController() {
+        guard let view = masterViewController?.view else {
+            return
+        }
+        
+        addViewToContainer(view)
+    }
+    
+    private func setFollowerController() {
+        guard let view = followerViewController?.view else {
+            return
+        }
+        
+        addViewToContainer(view)
+    }
+    
+    private func addViewToContainer(_ view: UIView) {
+        containerView.subviews.forEach {
+            $0.removeFromSuperview()
+        }
+        
+        view.translatesAutoresizingMaskIntoConstraints = false
+        containerView.addSubview(view)
+        view.bindToSuperview()
```

Comment:
What you did here is basically an implementation of tab bar controller. Let's put a `UITabBarController` on the top level (instead of 2 child view controllers) and manage it instead. Just hide a tab bar there.